### PR TITLE
Drop Python 2 support

### PR DESCRIPTION
**Pull request by andrewshadura**
It’s dead, Jim.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -64,7 +63,6 @@ def read(filename):
         'Development Status :: 4 - Beta',
         'Environment :: X11 Applications :: GTK',
         'License :: OSI Approved :: GNU General Public License (GPL)',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
```
Comment (mgedmin) [inline setup.py:68]: Would you mind also adding `python_requires='>= 3.5',` after the classifiers?

File: src/gtimelog/timelog.py [hunk 2]
```diff
@@ -1134,15 +1131,8 @@ class CSVWriter(object):
...
-        def writerow(self, row):
-            self._writer.writerow(row)
-    else:  # pragma: PY2
-        def writerow(self, row):
-            self._writer.writerow([
-                s.encode('UTF-8') if isinstance(s, unicode) else s  # noqa: F821
-                for s in row
-            ])
+    def writerow(self, row):
+        self._writer.writerow(row)
```
Comment (mgedmin) [inline src/gtimelog/timelog.py:1135]: The entire CSVWriter class can go away now, we can use `csv.writer` directly.

File: tox.ini [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-    py27,py35,py36,py37,py38,pypy,pypy3,flake8
+    py35,py36,py37,py38,pypy,pypy3,flake8
 
 [testenv]
 setenv =
     LC_ALL=C
 deps =
     freezegun
-    py27: mock
     pypy: mock
```
Comment (mgedmin) [inline tox.ini:10]: pypy is Python 2.7 too and should go away

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,5 @@
 language: python
 python:
-  - 2.7
   - 3.5
   - 3.6
   - 3.7
```
Comment (mgedmin) [inline .travis.yml:5]: There's also pypy which is 2.7 and should go away too.

**Review by mgedmin — CHANGES_REQUESTED** [review]
Some slight changes needed (python_requires, pypy2 drop), and it would be good to have a changelog entry about "Dropped Python 2.7 support" in CHANGES.rst.  Otherwise looks good!

File: src/gtimelog/tests/test_timelog.py [hunk 5]
```diff
@@ -11,18 +11,10 @@
 import time
 import unittest
 import sys
```
Comment (mgedmin) [inline src/gtimelog/tests/test_timelog.py:13]: The flake8 job failed with

> src/gtimelog/timelog.py:13:1: F401 'sys' imported but unused

so this line needs to go away too.

File: setup.py (hunk 1 above)
Comment (andrewshadura) [reply setup.py:68]: Sure.

File: src/gtimelog/timelog.py (hunk 2 above)
Comment (andrewshadura) [reply src/gtimelog/timelog.py:1135]: Indeed.

File: src/gtimelog/tests/test_timelog.py (hunk 5 above)
Comment (andrewshadura) [reply src/gtimelog/tests/test_timelog.py]: Hmm, apparently not quite.

Comment (andrewshadura) [reply src/gtimelog/tests/test_timelog.py]: Doctests use it, so it’s better to keep it. However, `# noqa: F401` doesn’t seem to work :disappointed:

Comment (mgedmin) [reply src/gtimelog/tests/test_timelog.py]: Mea culpa, I attached this comment on the wrong file!

It's not src/gtimelog/tests/test_timelog.py that needs to drop the `import sys`, but src/gtimelog/timelog.py.

File: src/gtimelog/tests/test_timelog.py [hunk 6]
```diff
@@ -10,19 +10,11 @@
 import textwrap
 import time
 import unittest
-import sys
-try:
-    from cStringIO import StringIO
-except ImportError:
-    from io import StringIO
+import sys  # noqa: F401  (doctests use sys.stdout)
```
Comment (mgedmin) [inline src/gtimelog/tests/test_timelog.py:13]: Sorry sorry sorry this is the wrong file, it doesn't need the `# noqa`

File: src/gtimelog/timelog.py [hunk 7]
```diff
@@ -17,9 +17,6 @@
 from operator import itemgetter
```
Comment (mgedmin) [inline src/gtimelog/timelog.py:17]: This is the right file with the superfluous `import sys` at the top.

(What a coincidence, it's on line 13 on both files, or I would've noticed sooner!)

File: CHANGES.rst [hunk 8]
```diff
@@ -18,6 +18,7 @@ Changelog
 
 - Add Python 3.8 support.
 
+- Drop Python 2.7 support.
```
Comment (mgedmin) [inline CHANGES.rst:22]: Minor: I like to keep two blank lines between ReST sections.

**Review by mgedmin — APPROVED** [review]

**Comment by mgedmin** [comment]
And soon Python 3.5 will EOL, and I will be able to use f-strings **everywhere**.

(Oh wait gettext won't work with that, *sigh*.  At least I'll have ordered dictionaries.)

**Pull request merged**